### ISSUE TITLE
LOG(fatal) is supposed to throw an exception but it never throws because of abort from fclose(stderr)

### ISSUE THREAD

**Issue by YanzhaoW** [FairRootGroup/FairRoot#1463]
**Describe the bug**
https://github.com/FairRootGroup/FairRoot/blob/fa7108b45684c729c2e88b12d7fbf4bbb5a4dc65/fairtools/FairLogger.cxx#L192-L203

Here the program somehow call the `abort` when calling `fclose(stderr)` in the if statement. Therefore, it never reaches to the throwing line.

**To Reproduce**
Add a printout just above the throw statement and the printout message never gets printed.

**Expected behavior**
Let it throw

**Comment by YanzhaoW** [comment]
Yeah, I agree. But would this be a huge breaking change? Many programs would not be terminated in the emergency, which could cause lots of segmentation fault.

**Comment by rbx** [comment]
I did a quick test with `LOG(fatal)` at (https://github.com/FairRootGroup/FairRoot/blob/dev/examples/MQ/serialization/data_generator/dataGenerator.cxx#L94) and it throws the exception as expected (which would itself lead to SIGABRT if uncaught).

Could you post a stacktrace to see a bit more detail?

I wonder why the close(stderr) is needed at all. Perhaps it is aborting because it is closed already at that point?
